src/cm1.F

```diff
@@ -2207,6 +2209,8 @@ program cm1
       pdata = 0.0
       allocate(   ploc(nparcels,  3   ) )
       ploc = 0.0
+      allocate(  pdata_locind(nparcels,3) )
+      pdata_locind = -100
```

Comment:
There are multiple locations for the setting of -100 to be an undefined value.  Could we do this in a single location so that it does not accidentally get changed to something else and then it breaks the code?

Reply:
Do you mean that some variables/arrays are uninitialized in CM1? Maybe we could write an `initialization` subroutine and always call this subroutine when we want to initialize a scalar/1-D/2-D/3-D array? In this way we could set `-100` in that subroutine once for the whole program.

Reply:
I would suggest that you create a parameter with a name like "undefined_index" in a single location in the source code and then you test against it in the parcel.F90 file.

Reply:
Thanks @johnmauff for your detailed suggestion. I have added a parameter named `undefined_index` in the `constants.F` source code and defined it as `-100`. Other codes are updated correspondingly.